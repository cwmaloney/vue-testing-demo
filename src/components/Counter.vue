<template>
	<div class="counter">
	  {{ count }}
	  <button id="increment-button" @click="increment">Increment</button>
	  <button id="decrement-button" @click="decrement">Decrement</button>
	</div>
</template>

<script>
export default {
  data () {
    return {
      count: 0
    };
  },

  methods: {
    increment () {
      this.count++;
    },
    decrement () {
      this.count--;
    }
  }
}
</script>

<style lang="css" scoped>
.counter {
  border: solid;
  margin: 10px 0;
  padding: 5px;
}
</style>
