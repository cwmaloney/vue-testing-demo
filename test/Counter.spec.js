import { shallowMount } from '@vue/test-utils'
import Counter from '../src/components/Counter.vue'

//import { extendExpect, attachSnapshot, detachSnapshot } from "./SnapshotUtilities.js"

// import snapshot from "snap-shot-it";

//extendExpect();

describe('Counter.vue', () => {

  //beforeEach(attachSnapshot);

  //afterEach(detachSnapshot);

  it('increments count when button is clicked', () => {
    const wrapper = shallowMount(Counter);
    wrapper.find("#increment-button").trigger('click');
    expect(wrapper.find('div').text()).toContain('1');
  });

  it('decrements count when button is clicked', () => {
    const wrapper = shallowMount(Counter);
    wrapper.find("#decrement-button").trigger('click');
    expect(wrapper.find('div').text()).toContain('-1');
  });

  it('rendors counter', () => {
    const wrapper = shallowMount(Counter);
    expect(wrapper.html()).toMatchSnapshot();
    // snapshot(wrapper.html());
  });

});
