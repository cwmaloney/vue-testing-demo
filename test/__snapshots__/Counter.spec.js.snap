// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue rendors counter 1`] = `
<div class="counter">
  0
  <button id="increment-button">Increment</button>
  <button id="decrement-button">Decrement</button>
</div>
`;
